setup/base.go

```diff
@@ -311,6 +312,17 @@ func (b *BaseDendrite) SetupAndServeHTTP(
 		}
 	}
 
+	externalRouter.Handle("/health",
+		httputil.MakeExternalAPI("health", func(req *http.Request) util.JSONResponse {
+			return util.JSONResponse{
+				Code: http.StatusOK,
+				JSON: struct {
+					Versions bool `json:"success"`
+				}{true},
+			}
+		}),
+	).Methods(http.MethodGet)
```

Comment:
One problem I am seeing is that `setup/base.go` -> `SetupAndServeHTTP` isn't called in [`cmd/dendrite-demo-libp2p/main.go`](https://github.com/matrix-org/dendrite/blob/4a0461378ad0d0af2ba701fb07dea505be469c92/cmd/dendrite-demo-libp2p/main.go) like it is for the [monolith](https://github.com/matrix-org/dendrite/blob/4a0461378ad0d0af2ba701fb07dea505be469c92/cmd/dendrite-monolith-server/main.go#L161-L170) and [polylith](https://github.com/matrix-org/dendrite/blob/4a0461378ad0d0af2ba701fb07dea505be469c92/cmd/dendrite-polylith-multi/personalities/clientapi.go#L40-L44) binaries so the health check isn't being configured in the `libp2p` demo. This means that when you run `./build.sh && ./bin/dendrite-demo-libp2p`, http://localhost:8080/health isn't available.

To be clear, the current code works fine in the monolith/polylith. Just have a problem with the `libp2p` demo

@neilalexander Can we use the standard `SetupAndServeHTTP` function in `cmd/dendrite-demo-libp2p/main.go` instead of using our own custom [`http.ListenAndServe(httpBindAddr, httpRouter)`](https://github.com/matrix-org/dendrite/blob/4a0461378ad0d0af2ba701fb07dea505be469c92/cmd/dendrite-demo-libp2p/main.go#L216) call (adapt whatever is necessary)? It looks like we do some [`embed.Embed(...)` stuff](https://github.com/matrix-org/dendrite/blob/4a0461378ad0d0af2ba701fb07dea505be469c92/cmd/dendrite-demo-libp2p/main.go#L205) to the router which makes it different from using `SetupAndServeHTTP`

ex.
```go
httpAddr := config.HTTPAddress("http://" + httpBindAddr)
httpAPIAddr := httpAddr

base.Base.SetupAndServeHTTP(
	httpAPIAddr, // internal API
	httpAddr,    // external API
	nil, nil,    // TLS settings
)
```

Reply:
To be honest I'm not terribly worried about the P2P demos here, as long as this does the right thing for monolith and polylith.